### PR TITLE
add trigger,trigger handler with tests

### PR DESCRIPTION
**Pull request by tgromad**
Add Trigger with tests to check for Season duplicates in TV_Serie

### CODE REVIEW

File: force-app/main/default/classes/TestDataFactory.cls [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        season.Sesaon_Name__c = name;
+        season.TV_Serie__c = tvserie;
+        return season;
+    }
+
+    public static TV_Serie__c getSimpleTvSerie(String url, String description) {
+        TV_Serie__c tvserie = new TV_Serie__c();
+        tvserie.Logo_url__c = url;
+        tvserie.Description__c = description;
+        tvserie.Year_of_production__c = '1999';
```
Comment (mlrogala) [inline force-app/main/default/classes/TestDataFactory.cls:31]: I think there should be also a tv serie name ;)

File: force-app/main/default/classes/DuplicateSeasonHandler.cls [hunk 2]
```diff
@@ -0,0 +1,47 @@
+public with sharing class DuplicateSeasonHandler {
+
+
+    public static void checkDuplicates(List<Season__c> seasons) {
+ 		
+        System.debug('###Season__c from trigger: ' + seasons);      
+        
+        List<Id> tvSerieIds = new List<Id>();
```
Comment (mlrogala) [inline force-app/main/default/classes/DuplicateSeasonHandler.cls]: use Set instead of List

File: force-app/main/default/classes/DuplicateSeasonHandler.cls [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        for(Season__c s : seasons) {
+            tvSerieIds.add(s.TV_Serie__c);
+        }
+        System.debug('###TV_Serie_Id from trigger: ' + tvSerieIds);
+        
+		Map<String,Id> mapSeasons = new Map<String,Id>();
+        // takes seasons that tv series id are in input values
+        List<Season__c> seasonsInDB = [SELECT Id,Sesaon_Name__c,TV_Serie__c FROM Season__c where TV_Serie__c IN :tvSerieIds];
```
Comment (mlrogala) [inline force-app/main/default/classes/DuplicateSeasonHandler.cls]: before that select check if tvSerieIds is not empty

File: force-app/main/default/classes/DuplicateSeasonHandler.cls [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        
+        // create a map where key is a season name and value is tv serie
+        if(!seasonsInDB.isEmpty()) {
+        System.debug('###Taking seasons from DB');
+            for(Season__c season : seasonsInDB) {
+            mapSeasons.put(season.Sesaon_Name__c,season.TV_Serie__c);
```
Comment (mlrogala) [inline force-app/main/default/classes/DuplicateSeasonHandler.cls]: what if for 2 different tv series we will have seasons with the same name?
Create Map<String,List<Id>>
as String key: season name
as list<Id>: ids of tv series which are related with season with given name

File: force-app/main/default/classes/DuplicateSeasonHandler.cls [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        
+/* Iterate on seasons by looking if map key(Season_Name) have same value(TV_Serie) 
+ * The Name of seasons should be unique in same TV_Serie that's why the Name is a key
+ * The Map contains the values from database
+ * If our Input (seasons) Name is already in our mapSeassons(key) and the values (TV_Serie Id) are same
+ * call Error message
+ */
+        if(!mapSeasons.keySet().isEmpty()) {
+        	for(Season__c season : seasons) {
+            	if(mapSeasons.get(season.Sesaon_Name__c) == season.TV_Serie__c){
```
Comment (mlrogala) [inline force-app/main/default/classes/DuplicateSeasonHandler.cls]: if you change Map<String,Id> into Map<String,List<Id>> replace == by contains() method

**Review by mlrogala — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/TestDataFactory.cls (hunk 1 above)
Comment (tgromad) [reply force-app/main/default/classes/TestDataFactory.cls:31]: Done

File: force-app/main/default/classes/DuplicateSeasonHandler.cls (hunk 2 above)
Comment (tgromad) [reply force-app/main/default/classes/DuplicateSeasonHandler.cls]: Done

File: force-app/main/default/classes/DuplicateSeasonHandler.cls (hunk 3 above)
Comment (tgromad) [reply force-app/main/default/classes/DuplicateSeasonHandler.cls]: Done

File: force-app/main/default/classes/DuplicateSeasonHandler.cls (hunk 4 above)
Comment (tgromad) [reply force-app/main/default/classes/DuplicateSeasonHandler.cls]: Done
The Map is now<Id,List>
Key : TV_Serie_Id
as list: Names of Series

TV_Serie_Id is always Unique that's why i set it as key ;)

File: force-app/main/default/classes/DuplicateSeasonHandler.cls (hunk 5 above)
Comment (tgromad) [reply force-app/main/default/classes/DuplicateSeasonHandler.cls]: Done

File: force-app/main/default/classes/DuplicateSeasonHandler.cls [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        System.debug('###Taking seasons from DB');
+            for(Id tvId : tvSerieIds) {
+                for(Season__c season : seasonsInDB) {
+                    if(season.TV_Serie__c == tvId) {
+                        SeasonsName.add(season.Sesaon_Name__c);
+                        System.debug('### season name '+season.Sesaon_Name__c);
+                    	}
+                    }
+                    System.debug('### seasonsName list '+SeasonsName.size());
+                    mapSeasons.put(tvId,SeasonsName.clone());
```
Comment (mlrogala) [inline force-app/main/default/classes/DuplicateSeasonHandler.cls]: you don't have to clone this list. just put it into map

File: force-app/main/default/classes/DuplicateSeasonHandler.cls [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                    SeasonsName.clear();
```
Comment (mlrogala) [inline force-app/main/default/classes/DuplicateSeasonHandler.cls]: If you create SeasonsName list not before if(!seasonsInDB.isEmpty()) but in the first loop (between for(Id tvId..) and for(Season__c season...)), you will not have to clear it at the end ;)

File: force-app/main/default/classes/DuplicateSeasonHandler.cls [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+        // create list for season names
+        List<String> SeasonsName = new List<String>();
+
+        if(!seasonsInDB.isEmpty()) {
+        System.debug('###Taking seasons from DB');
+            for(Id tvId : tvSerieIds) {
+                for(Season__c season : seasonsInDB) {
+                    if(season.TV_Serie__c == tvId) {
+                        SeasonsName.add(season.Sesaon_Name__c);
```
Comment (mlrogala) [inline force-app/main/default/classes/DuplicateSeasonHandler.cls:26]: there is a typo in API name for Season Name (it's Sesaon_Name__c). Is it the same on org? If yes - leave it here like it is. If not - change it in trigger.

**Review by mlrogala — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/DuplicateSeasonHandler.cls (hunk 8 above)
Comment (tgromad) [reply force-app/main/default/classes/DuplicateSeasonHandler.cls:26]: Unfortunately it's same as in org

File: force-app/main/default/classes/DuplicateSeasonHandler.cls (hunk 6 above)
Comment (tgromad) [reply force-app/main/default/classes/DuplicateSeasonHandler.cls]: Done

File: force-app/main/default/classes/DuplicateSeasonHandler.cls (hunk 7 above)
Comment (tgromad) [reply force-app/main/default/classes/DuplicateSeasonHandler.cls]: Done

**Review by mlrogala — APPROVED** [review]
